scripts/deep_learning/hierarchy/LaterTier/TrainLaterTierModel_showers.ipynb

```diff
@@ -0,0 +1,897 @@
+    "trainingCutT_test = trainingCutT_test[passTrainingCuts_test]     # Don't really need to do this as we're not using them again\n",
+    "\n",
+    "# Mask the truth... shape=(nEntries, nLinks)\n",
+    "y_test = y_test[np.column_stack((passTrainingCuts_test, passTrainingCuts_test))].reshape(-1, nLinks)\n",
+    "\n",
+    "# Mask the variable... shape=(nEntries, nVariables)\n",
+    "variables_test = variables_test[[[entry] * nVariables for entry in passTrainingCuts_test]].reshape(-1, nVariables)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9852ed01",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "###########################################################\n",
+    "# Check shapes\n",
+    "###########################################################\n",
+    "\n",
+    "print('variables_train.shape:', variables_train.shape)\n",
+    "print('variables_test.shape:', variables_test.shape)\n",
+    "print('y_train.shape:', y_train.shape)\n",
+    "print('y_test.shape:', y_test.shape)\n",
+    "print('trueParentChildLink_train.shape:', trueParentChildLink_train.shape)\n",
+    "print('trueParentChildLink_test.shape:', trueParentChildLink_test.shape)\n",
+    "print('trueChildVisibleGeneration_train.shape:', trueChildVisibleGeneration_train.shape)\n",
+    "print('trueChildVisibleGeneration_test.shape:', trueChildVisibleGeneration_test.shape)\n",
+    "print('trainingCutSep_train.shape:', trainingCutSep_train.shape)\n",
+    "print('trainingCutSep_test.shape:', trainingCutSep_test.shape)\n",
+    "print('trainingCutDoesConnect_train.shape:', trainingCutDoesConnect_train.shape)\n",
+    "print('trainingCutDoesConnect_test.shape:', trainingCutDoesConnect_test.shape)\n",
+    "print('trainingCutL_train.shape:', trainingCutL_train.shape)\n",
+    "print('trainingCutL_test.shape:', trainingCutL_test.shape)\n",
+    "print('trainingCutT_train.shape:', trainingCutT_train.shape)\n",
+    "print('trainingCutT_test.shape:', trainingCutT_test.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bf470a8a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "###########################################################\n",
+    "# Branch weights!\n",
+    "###########################################################\n",
+    "\n",
+    "# Calculate branch weights\n",
+    "nTrue = np.count_nonzero(y_train == 1)\n",
+    "nBackground = np.count_nonzero(y_train == 0)\n",
+    "nWrongOrientation = np.count_nonzero(y_train == 2)\n",
+    "\n",
+    "maxLinks = max(nTrue, nBackground, nWrongOrientation)\n",
+    "\n",
+    "true_branch_weight = float(maxLinks)/float(nTrue)\n",
+    "background_branch_weight = float(maxLinks)/float(nBackground)\n",
+    "wrong_orientation_branch_weight = float(maxLinks)/float(nWrongOrientation)\n",
+    "\n",
+    "print(nTrue)\n",
+    "print(background_branch_weight)\n",
+    "print(wrong_orientation_branch_weight)\n",
```

Comment:
If you want to print these variable values, add some text indicating what's being printed.

Reply:
removed, the relevant info is printed below